ts/src/hashkey.ts

```diff
@@ -0,0 +1,4313 @@
+                request['type'] = 'LIMIT';
+            }
+            if (market !== undefined) {
+                request['symbol'] = market['id'];
+            }
+            response = await this.privateDeleteApiV1FuturesOrder (this.extend (request, params));
+            //
+            //     {
+            //         "time": "1722432302919",
+            //         "updateTime": "1722432302925",
+            //         "orderId": "1742282868229463040",
+            //         "clientOrderId": "1722432301670",
+            //         "symbol": "ETHUSDT-PERPETUAL",
+            //         "price": "4000",
+            //         "leverage": "5",
+            //         "origQty": "10",
+            //         "executedQty": "0",
+            //         "avgPrice": "0",
+            //         "marginLocked": "0",
+            //         "type": "LIMIT_MAKER",
+            //         "side": "SELL_CLOSE",
+            //         "timeInForce": "GTC",
+            //         "status": "NEW",
+            //         "priceType": "INPUT",
+            //         "isLiquidationOrder": false,
+            //         "indexPrice": "0",
+            //         "liquidationType": ""
+            //     }
+            //
+        } else {
+            throw new NotSupported (this.id + ' ' + methodName + '() is not supported for ' + marketType + ' type of markets');
+        }
+        return this.parseOrder (response);
+    }
+
+    async cancelAllOrders (symbol: Str = undefined, params = {}) {
+        /**
+         * @method
+         * @name hashkey#cancelAllOrders
+         * @description cancel all open orders
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/cancel-all-open-orders
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/batch-cancel-futures-order
+         * @param {string} symbol unified market symbol, only orders in the market of this symbol are cancelled when symbol is not undefined
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.side] 'buy' or 'sell'
+         * @returns {object} response from exchange
+         */
+        // Does not cancel trigger orders. For canceling trigger order use cancelOrder() or cancelOrders()
+        const methodName = 'cancelAllOrders';
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' ' + methodName + '() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+        };
+        let side: Str = undefined;
+        [ side, params ] = this.handleOptionAndParams (params, 'cancelAllOrders', 'side');
+        if (side !== undefined) {
+            request['side'] = side;
+        }
+        if (market['spot']) {
+            return await this.privateDeleteApiV1SpotOpenOrders (this.extend (request, params));
```

Comment:
cancelAllOrders should return the unified type `Order[]` (check unified implementations)